megamek/src/megamek/client/bot/princess/RankedPath.java

```diff
@@ -72,13 +72,7 @@ public int compareTo(RankedPath p) {
         if (p.rank < rank) {
             return 1;
         }
-        if (path.getKey().hashCode() < p.path.getKey().hashCode()) {
-            return -1;
-        }
-        if (path.getKey().hashCode() > p.path.getKey().hashCode()) {
-            return 1;
-        }
-        return 0;
+        return Integer.compare(path.getHexesMoved(), p.path.getHexesMoved());

```

Comment:
I added the `hexes moved` as a second level of compare to break ties.

Reply:
Hexes moved seems like the better option in the general case, as that drives TMM.